resources/puppetlabs/lein-ezbake/template/global/ext/fpm.rb

```diff
@@ -351,18 +351,42 @@
 shared_opts << "--architecture all"
 
 options.replaces.each do |pkg, version|
+  # Strip the surrounding quotes since we add them in a certain way here.
+  # We should probably just fix this in the core code by being smarter with
+  # as-ruby-literaly, but someone more familiar with Clojure can do that part.
+  pkg = pkg.delete_prefix("'").delete_suffix("'")
+  version = version.delete_prefix("'").delete_suffix("'") unless version.nil?
   if options.output_type == 'rpm'
-    fpm_opts << "--replaces '#{pkg} <= #{version}-1'"
-    fpm_opts << "--conflicts '#{pkg} <= #{version}-1'"
+    val = if version.nil? || version.empty?
+            "'#{pkg}'"
+          else
+            "'#{pkg} <= #{version}-1'"
+          end
+    fpm_opts << "--replaces #{val}"
+    fpm_opts << "--conflicts #{val}"
```

Comment:
FYI: this is frowned upon in RPM land. They really should have versions and rpmlint will yell at you because it doesn't allow going back.

Wouldn't it be better to declare it always conflicts with the package (regardless of version) and only declare it replaces older versions?

Here I rely on the `%{version}` RPM macro that I hope will work with FPM
```suggestion
    version = (version.nil? || version.empty?) ? '%{version}' : "#{version}-2"
    fpm_opts << "--replaces '#{pkg} < #{version}"
    fpm_opts << "--conflicts '#{pkg}'"
```